src/composite_core.jl

```diff
@@ -0,0 +1,124 @@
+struct PrimalAdditionFailedException{P} <: Exception
+    primal::P
+    differential::Composite{P}
+    original::Exception
```

Comment:
Could the `original field be parameterized` `E <: Exception`?

Reply:
What would the advantage be?

Reply:
unboxed type

Reply:
all other fields are concrete, this would allow further optimization

Reply:
FWIW `Composite{P}` also isn't a concrete type, although I'm not sure, it really matters here. Might even be beneficial to avoid overspecialization.

Reply:
It is an exception and one that really should not be caught because it indicated a programmer mistake